netmaster/objApi/apiController.go

```diff
@@ -860,6 +1015,79 @@ func (ac *APIController) NetworkDelete(network *contivModel.Network) error {
 	return nil
 }
 
+// NetprofileCreate creates the network rule
+func (ac *APIController) NetprofileCreate(netProfile *contivModel.Netprofile) error {
+	log.Infof("Received NetprofileCreate: %+v", netProfile)
+
+	// Check if the tenant exists
+	if netProfile.TenantName == "" {
+		return core.Errorf("Invalid tenant name")
+	}
+
+	if netProfile.Burst > 0 && netProfile.Burst < 2 {
+		return core.Errorf("Invalid Burst size. burst size > 1500 bytes")
+	}
+
+	tenant := contivModel.FindTenant(netProfile.TenantName)
+	if tenant == nil {
+		return core.Errorf("Tenant not found")
+	}
+
+	// Setup links & Linksets.
+	modeldb.AddLink(&netProfile.Links.Tenant, tenant)
+	modeldb.AddLinkSet(&tenant.LinkSets.NetProfiles, netProfile)
+
+	// Save the tenant in etcd - This writes to etcd.
+	err := tenant.Write()
+	if err != nil {
+		log.Errorf("Error updating tenant state(%+v). Err: %v", tenant, err)
+		return err
+	}
+
+	return nil
+}
+
+// NetprofileUpdate updates the netprofile
+func (ac *APIController) NetprofileUpdate(profile, params *contivModel.Netprofile) error {
+	log.Infof("Received NetprofileUpdate: %+v, params: %+v", profile, params)
+
+	if params.Burst > 0 && params.Burst < 2 {
+		return core.Errorf("Invalid Burst size. burst size must be > 1500 bytes")
+	}
+	profile.Bandwidth = params.Bandwidth
+	profile.DSCP = params.DSCP
+	profile.Burst = params.Burst
+
+	for key := range profile.LinkSets.EndpointGroups {
+		// Find the corresponding epg
+		epg := contivModel.FindEndpointGroup(key)
+		if epg == nil {
+			return core.Errorf("EndpointGroups not found")
+		}
+		//using the epg structure,find the epg key
+		epgkey := mastercfg.GetEndpointGroupKey(epg.GroupName, epg.TenantName)
+		if epgkey == "" {
+			log.Errorf("Error finding endpointGroup")
+		}
+		err := master.UpdateEndpointGroup(params.Bandwidth, epgkey, params.DSCP, params.Burst)
+		if err != nil {
+			log.Errorf("Error updating the EndpointGroups: %s. Err: %v", epgkey, err)
+		}
+	}
+	return nil
+}
+
+// NetprofileDelete deletes netprofile
+func (ac *APIController) NetprofileDelete(profile *contivModel.Netprofile) error {
+	log.Infof("received NetprofileDelete: %+v", profile)
+
+	// Check if any endpoint group is using the network policy
+	if len(profile.LinkSets.EndpointGroups) != 0 {
+		return core.Errorf("NetProfile is being used")
+	}
+	return nil
```

Comment:
Please remove the net profile from tenant linkset
Also add a unit test case to make sure tenant can not be deleted if it has a net profile and tenant can be deleted after net profile is removed..